src/main/java/org/cactoos/func/Equality.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import org.cactoos.Bytes;
+import org.cactoos.Scalar;
+
+/**
+ * Equality.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Yegor Bugayenko (yegor256@gmail.com)
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @param <T> Type of input
+ * @since 0.12
+ */
+public final class Equality<T extends Bytes> implements Scalar<Integer> {
+
+    /**
+     * Left.
+     */
+    private final T lft;
+
+    /**
+     * Right.
+     */
+    private final T rght;
+
+    /**
+     * Ctor.
+     * @param left The first object to compare
+     * @param right The second object to compare
+     */
+    public Equality(final T left, final T right) {
+        this.lft = left;
+        this.rght = right;
+    }
+
+    @Override
+    public Integer value() throws Exception {
+        final byte[] left = this.lft.asBytes();
+        final byte[] right = this.rght.asBytes();
+        final int max = Math.max(left.length, right.length);
+        int result = 0;
+        for (int idx = max; idx > 0; --idx) {
+            byte first = 0;
+            if (idx <= left.length) {
```

Comment:
@fabriciofx Looks like code can be more optimal. You can remove `if` statements from the loop. Get a longer array and check its last bytes is not zero. After that, you can compare a shorter array with remaining part of the longer one with only one `if` statement.